ui/src/features/get-test-reports.js

```diff
@@ -53,16 +53,18 @@ class getTests extends React.Component {
         this.setState({rerunJob: job});
     };
 
-    filterByFavoriteState = () => {
+    filterFavorites = () => {
         const {onlyFavorites, sortedReports} = this.state;
-        const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
-        this.setState({sortedReports: filteredReports, sortHeader: ''});
+        if (onlyFavorites) {
+            const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
```

Comment:
Can be changed to
const filteredReports = sortedReports.filter((report) => (report.is_favorite));